consentconstants/purposes.go

```diff
@@ -39,3 +40,87 @@ const (
 	// for you in other contexts, i.e. on other service, such as websites or apps, over time.
 	Measurement Purpose = 5
 )
+
+// TCF 2.0 Purposes
+const (
+	// Cookies, device identifiers, or other information can be stored or accessed on your device for the purposes presented to you.
+	// Vendors can:
+	//   * Store and access information on the device such as cookies and device identifiers presented to a user.
+	// Reuse InfoStorageAccess above
+
+	// Ads can be shown to you based on the content you are viewing, the app you are using, your approximate location, or your device type.
+	// To do basic ad selection vendors can:
+	//   * Use real-time information about the context in which the ad will be shown, to show the ad, including information about the content and
+	//     the device, such as: device type and capabilities, user agent, URL, IP address
+	//   * Use a user's non-precise geolocation data
+	//   * Control the frequency of ads shown to a user.\n* Sequence the order in which ads are shown to a user.
+	//   * Prevent an ad from serving in an unsuitable editorial (brand-unsafe) context
+	// Vendors cannot:
+	//   * Create a personalised ads profile using this information for the selection of future ads.
+	//   * N.B. Non-precise means only an approximate location involving at least a radius of 500 meters is permitted.
+	BasicAdserving Purpose = 2
```

Comment:
Does it make sense to name these something like `TCF2BasicAdserving`?

Reply:
IAB wants to deprecate TCF1 as fast as possible, so we should be able to pull all the old names fairly quickly.